### PR TITLE
Persist and load embedding configuration changes using DynamoDB

### PR DESCRIPTION
**Pull request by marcellp**
# Background
#### Link to issue
https://biomage.atlassian.net/browse/BIOMAGE-450

#### Link to staging deployment URL
https://ui-tup1zojf7s7jzginx8tzmtq1zo.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/inframock/pull/18

#### Anything else the reviewers should know about the changes here
N/A

# Changes
### Code changes
Added functionality for loading and saving experiment processing settings to/from DynamoDB.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock
- [x] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [x] Relevant Github READMEs updated
- [x] Relevant wiki pages created/updated

### Approvers
- [x] Approved by a member of the core engineering team
- [x] (UX changes) Approved by @vickymorrison 

### Just before merging:
- [x] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [x] Photo of a cute animal attached to this PR
image

### CODE REVIEW

**Comment by iguanaonmystack** [comment]
looks good. I like the 1.5s debounce stuff that, if I understand correctly, stops every user input/keypress from trying to save immediately

**Comment by marcellp** [comment]
> looks good. I like the 1.5s debounce stuff that, if I understand correctly, stops every user input/keypress from trying to save immediately

Correct. The debounce makes sure that the function wrapped in the debounce is not called until a certain amount of time has elapsed since the last time the debounce was called.

**Comment by marcellp** [comment]
Staging available at https://ui-tup1zojf7s7jzginx8tzmtq1zo.scp-staging.biomage.net/.

File: src/redux/actionTypes/experimentSettings.js [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
  * Updates the processing component's configuration.
  */
 const EXPERIMENT_SETTINGS_PROCESSING_UPDATE = `${EXPERIMENT_SETTINGS}/updateProcessing`;
 
+/**
+ * Saves the current state of the configuration to DynamoDB.
+ */
+const EXPERIMENT_SETTINGS_PROCESSING_SAVE = `${EXPERIMENT_SETTINGS}/saveProcessing`;
+
 const EXPERIMENT_SETTINGS_PROCESSING_ERROR = `${EXPERIMENT_SETTINGS}/errorProcessing`;
```
Comment (ivababukova) [inline src/redux/actionTypes/experimentSettings.js:18]: are we not going to use this `EXPERIMENT_SETTINGS_PROCESSING_ERROR` in these changes if dynamodb returns an error?

File: src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx [hunk 2]
```diff
@@ -20,20 +22,55 @@ const MIN_DIST_TEXT = 'This controls how tightly the embedding is allowed compre
...
   const { method: embeddingMethod } = data.embeddingSettings;
   const { umap: umapSettings, tsne: tsneSettings } = data.embeddingSettings.methodSettings;
   const { louvain: louvainSettings } = data.clusteringSettings.methodSettings;
 
-  const updateSettings = (diff) => dispatch(updateProcessingSettings(
-    experimentId,
-    FILTER_UUID,
-    diff,
-  ));
+  console.log(data);
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx]: remove this

File: src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx [hunk 3]
```diff
@@ -52,9 +89,11 @@ const CalculationConfig = (props) => {
         <InputNumber
```
Comment (ivababukova) [inline src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx:90]: I just noticed that the number does not change when we click the up and down arrows just next to the "minimum distance" number - could you take a look and fix it? If the arrows are a pain to fix, we can remove them too - I can see that on the wireframes they are not there.

File: src/redux/actionTypes/experimentSettings.js (hunk 1 above)
Comment (marcellp) [reply src/redux/actionTypes/experimentSettings.js:18]: We probably should, good point Iva.

File: src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx (hunk 3 above)
Comment (marcellp) [reply src/pages/experiments/[experimentId]/data-processing/configure-embedding/components/CalculationConfig.jsx:90]: Fixed.

**Review by ivababukova — APPROVED** [review]

**Pull request merged**